### PR TITLE
add a function to X2TacticalGameRuleset::CreateTacticalGame to allow …

### LINKED ISSUE #450: Give SitReps access to Tactical StartState
add a function to `X2TacticalGameRuleset::CreateTacticalGame` to allow SitReps to modify the StartState in the same way that Strategy Cards do

### PR DESCRIPTION
**Pull request by statusNone**
closes #450

### CODE REVIEW

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/X2SitRepEffect_ModifyTacticalStartState.uc [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class X2SitRepEffect_ModifyTacticalStartState extends X2SitRepEffectTemplate;
```
Comment (robojumper) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/X2SitRepEffect_ModifyTacticalStartState.uc]: Maybe a comment or class name that indicates that this class is new?

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/X2TacticalGameRuleset.uc [hunk 2]
```diff
@@ -2890,8 +2923,9 @@ Begin:
 	WorldInfo.MyLocalEnvMapManager.SetEnableCaptures(TRUE);
 
 	ReleaseScriptLog("X2TacticalGameRuleset: Finished Generating Map");
-
+	
 	ApplyResistancePoliciesToStartState( );
+	ApplySitRepEffectsToStartState();  // CHL issue #450
 	ApplyDarkEventsToStartState( );
```
Comment (robojumper) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/X2TacticalGameRuleset.uc]: 1. There's a stray whitespace.
2. Order? Is there a reason this is between Resistance Policies and Dark Events?

Comment (Xymanek) [reply X2WOTCCommunityHighlander/Src/XComGame/Classes/X2TacticalGameRuleset.uc]: The implementation was based on bringing functionality available for resistance orders but not for sitreps to latter. Considering that DEs/orders are already working and are done here i do not see a need to change the positioning/order.

If such need ever arises, we can just add another delegate to the template

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/X2SitRepEffect_ModifyTacticalStartState.uc [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+//
+//---------------------------------------------------------------------------------------
+// File added to the CHL this is a non-base game class
+//---------------------------------------------------------------------------------------
+
+class X2SitRepEffect_ModifyTacticalStartState extends X2SitRepEffectTemplate;
+
+var Delegate<ModifyTacticalStartStateDelegate> ModifyTacticalStartStateFn;
+
+delegate ModifyTacticalStartStateDelegate(XComGameState StartState);
```
Comment (Xymanek) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/X2SitRepEffect_ModifyTacticalStartState.uc]: I think it makes sense to override the validate function to check that the delegate has been set

**Pull request merged**